---
layout: page
title: Statistics
description: Leveraging multiomics to discover biology
img: assets/img/3.jpg
importance: 2
category: work
giscus_comments: true
related_publications: gorin_studying_2023,gorin_length_2023,gorin_distinguishing_2023,gorin_interpretable_2022,gorin_intrinsic_2020,gorin_protein_2020,carilli_biophysical_2023
---

Over the past decade, the increasing affordability of DNA sequencing has made RNA quantification more practical and spurred improvements in sequencing. Yet transcriptomics is only part of the sought-after comprehensive picture of single-cell biology. To hone understanding of biological regulation, it is necessary to quantify more modalities involved in information transfer: DNA and protein molecules, as well as transient and alternative RNA isoforms.

Simultaneously with advances in RNA sequencing, two factors have contributed to multimodal -omics: the development of analogous assays for chromatin accessibility and protein abundance, and the release of software infrastructure to quantify non-coding RNA molecules. If we run a scRNA-seq experiment, we get information about splicing for free. If we develop a slightly more sophisticated methodology, we can collect information about other modalities, either closer to gene regulation or to its effects.

Yet despite this profusion of experimental methods, the correct way to analyze such data is far from clear. Despite a 